src/services/profile.service.js

```diff
@@ -0,0 +1,123 @@
+const { ApiError } = require('../exceptions/api.error');
+const userService = require('../services/user.service');
+const emailService = require('../services/email.service');
+const bcrypt = require('bcrypt');
+const { checkRequired } = require('../utils/checkRequired');
+const { validatePassword } = require('../utils/validatePassword');
+const { validateEmail } = require('../utils/validateEmail');
+
+const passReset = async (accessToken) => {
+  const user = await userService.getByAccessToken(accessToken);
+
+  if (!user) {
+    throw ApiError.BadRequest('bad request', {
+      noUser: 'no user was found',
+    });
+  }
+
+  await emailService.sendPassConfirmationEmail(user.email, accessToken);
+
+  return userService.normalize(user);
+};
+
+const passResetConfirm = async (accessToken, newPass, newPassConfirmation) => {
+  const errors = {
+    accessToken: checkRequired(accessToken, 'access token'),
+    newPass: validatePassword(newPass),
+    newPassConfirmation: checkRequired(
+      newPassConfirmation,
+      'new password confirmation',
+    ),
+  };
+
+  if (errors.accessToken || errors.newPassConfirmation || errors.newPass) {
+    throw ApiError.BadRequest('bad request', errors);
+  }
+
+  const user = await userService.getByAccessToken(accessToken);
+
+  if (!user) {
+    throw ApiError.notFound();
+  }
+
+  if (!(newPass === newPassConfirmation)) {
+    throw ApiError.BadRequest('entered passwords are not equal');
+  }
+
+  user.password = await bcrypt.hash(newPass, 10);
+  user.save();
+
+  return user;
+};
+
+const changeName = async (newName, accessToken) => {
+  const newNameRequiredError = checkRequired(newName, 'new name');
+
+  if (newNameRequiredError) {
+    throw ApiError.BadRequest(newNameRequiredError);
+  }
+
+  const user = await userService.getByAccessToken(accessToken);
+
+  if (!user) {
+    throw ApiError.BadRequest('bad request', {
+      noUser: 'no user was found',
+    });
+  }
+
+  user.name = newName;
+  user.save();
+
+  return userService.normalize(user);
+};
+
+const changeEmail = async (newEmail, password, accessToken) => {
+  const errors = {
+    newEmail: validateEmail(newEmail),
+    password: checkRequired(password, 'password'),
+  };
+
+  if (errors.newEmail || errors.password) {
+    throw ApiError.BadRequest('bad request', errors);
+  }
+
+  const user = await userService.getByAccessToken(accessToken);
```

Comment:
Ensure that the user is verified before attempting to compare the password. If 'user' is null, this will cause an error. Consider adding a check to handle this case.